src/background.js

```diff
@@ -17,28 +17,32 @@
 import { init, stores } from './stores';
 import { triggerNotionAuthFlow } from './util/auth';
 import chromeStorageSyncStore from './util/chromeStorageSyncStore';
-import { getHaloCookies, getHaloUserInfo, getUserId, getUserOverview } from './util/halo';
+import { getInformation, getHaloUserInfo, getUserId, getUserOverview } from './util/halo';
 // no stores - code is not shared between background and popup
 
 const VERSION = chrome.runtime.getManifest().version;
-const COOKIE_KEY = 'halo_cookies';
 
 (async function () {
 	console.log(`${chrome.runtime.getManifest().name} v${VERSION}`);
 
 	console.log('initializing ApplicationStoreManager');
-	const initial_cookies = await getHaloCookies();
+	const { ['userId']: userId, ...cookies } = await getInformation();
+	console.log('fetched user id', userId);
+	console.log('fetched halo cookies', cookies);
 	await init([
 		chromeStorageSyncStore({ key: 'notion_info' }),
-		chromeStorageSyncStore({ key: COOKIE_KEY, initial_value: initial_cookies }),
-		chromeStorageSyncStore({ key: 'halo_info', initial_value: () => getHaloUserInfo({ cookie: initial_cookies }) }),
+		chromeStorageSyncStore({ key: 'halo_cookies', initial_value: cookies }),
+		chromeStorageSyncStore({
+			key: 'halo_info',
+			initial_value: async () => await getHaloUserInfo({ cookie: cookies }),
```

Comment:
```suggestion
			initial_value:  getHaloUserInfo({ cookie: cookies }),
```

the store is smart enough to await the initial_value

Reply:
oh that explains a lot, I was wondering how it was functioning in the first place.

Reply:
yeah it's essentially a custom Svelte store that syncs with the extension sync storage API... but since I built it custom, I added asynchronous support 😁 
https://github.com/elijaholmos/halo-notion-extension/blob/main/src/util/chromeStorageSyncStore.js